src/script/api/script_list.cpp

```diff
@@ -403,6 +403,43 @@ class ScriptListSorterItemDescending : public ScriptListSorter {
 
 
 
+bool ScriptList::SaveObject(HSQUIRRELVM vm)
+{
+	sq_pushstring(vm, "List");
+	sq_newtable(vm);
+	sq_pushinteger(vm, this->sorter_type);
+	sq_pushbool(vm, this->sort_ascending);
+	sq_rawset(vm, -3);
+	for (ScriptListMap::iterator iter = this->items.begin(); iter != this->items.end(); iter++) {
+		sq_pushinteger(vm, iter->first);
+		sq_pushinteger(vm, iter->second);
+		sq_rawset(vm, -3);
+	}
+	return true;
+}
+
+bool ScriptList::LoadObject(HSQUIRRELVM vm)
+{
+	if (sq_gettype(vm, -1) != OT_TABLE) return false;
+	sq_pushnull(vm);
+	while (SQ_SUCCEEDED(sq_next(vm, -2))) {
+		SQInteger key, value;
+		sq_getinteger(vm, -2, &key);
+		if (sq_gettype(vm, -1) == OT_BOOL) {
+			SQBool val;
+			sq_getbool(vm, -1, &val);
+			this->Sort(static_cast<SorterType>(key), val);
+			sq_pop(vm, 2);
+			continue;
+		}
+		sq_getinteger(vm, -1, &value);
+		this->AddItem(key, value);
+		sq_pop(vm, 2);
+	}
```

Comment:
Hmm, I would have expected sorter_type and sort_ascending to be read outside of the loop, and then maybe applied after the loop has finished reading the items.

Reply:
I'm not sure if the table data is sorted internally, or if it keeps the key order given on creation, or if it's totally random, so to be safe I just look for a bool value, which means it is the type/ascending item.
`AddItem` is not affected by sorter type or direction, it's only for iterating.

Reply:
Yeah, storing them inside the same table was a bad idea anyway, as the `type = ascending` could be overwritten if a list item had the same index as type.
I changed the save format into an array containing [type, ascending, table].

And during my tests I found out I was correct about the random order in tables, my test GS saves 
```{
    list = this.list
    tilelist = this.tilelist
}```
but during loading the first element was `tilelist`.